src/compiler/parser.ts

```diff
@@ -594,7 +599,7 @@ namespace ts {
             // If this is a javascript file, proactively see if we can get JSDoc comments for
             // relevant nodes in the file.  We'll use these to provide typing informaion if they're
             // available.
-            if (isJavaScript(fileName)) {
+            if (isSourceFileJavaScript(sourceFile)) {
                 addJSDocComments();

```

Comment:
The `addJSDocComments` function does a full AST walk. I think we could just do this in the binder.